### ISSUE TITLE
Extension properties, fields and operators

### ISSUE THREAD

**Issue by ghost** [dotnet/vblang#205]
A long time ago Extension Methods were added to C# and VB.
I propose to add extension properties and extension fields too.

**Comment by reduckted** [comment]
#73

**Comment by tverweij** [comment]
#114 would be easy if  Extension Operators are implemented.

**Comment by tverweij** [comment]
#204 might be unnecessary when everything can be extended by extensions, but it should then be possible to shadow an existing method, property, etc with an extension method.

**Comment by Bill-McC** [comment]
IIRC the design for extension methods is they are not to shadow methods in the type.  This is important so that importing a lubrary with extensions doesn't hijack and change methods intrinsic to the type. Extensions extend but not break encapsulation.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Comment by reduckted** [comment]
That's correct. You will never ever be able to shadow an instance method by using an extension method.

> When an in-scope instance method has a signature that is compatible with the arguments of a calling statement, the instance method is chosen in preference to any extension method. The instance method has precedence even if the extension method is a better match. 

From [Extension Methods, Instance Methods, and Properties](https://docs.microsoft.com/en-us/dotnet/visual-basic/programming-guide/language-features/procedures/extension-methods#extension-methods-instance-methods-and-properties).

**Comment by tverweij** [comment]
And I agree with it; as you describe, the side effects of shadowing extensions are unacceptable.

**Comment by tverweij** [comment]
dotnet/csharplang#192
dotnet/roslyn#6136

**Comment by KathleenDollard** [comment]
This is a great idea, and is being discussed. Happily, this has a Roslyn proposal. If that moves forward, extending :-) it to VB should be straightforward. 

More recent than [dotnet/roslyn#6136](dotnet/roslyn/https://github.com/dotnet/roslyn/issues/11159) is [dotnet/roslyn#11159](dotnet/roslyn/https://github.com/dotnet/roslyn/issues/11159)
